src/components/accounts/TransactionItem.tsx

```diff
@@ -0,0 +1,262 @@
+    if (unit === 'second') {
+        return 'less than a minute ' + suffix;
+    } else if (unit === 'minute' || unit === 'hour') {
+        return `${value} ${unit}${
+          value !== 1 ? 's' : ''
+      } ${suffix}`;
+    } else {
+      return `${formatTime(timestamp)} - ${formatDate(timestamp)}`;
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.icon}>
+        { transaction.type === TransactionType.Send && <OutgoingIcon /> }
+        { transaction.type === TransactionType.Receive && <IncomingIcon width={19} height={19} /> }
+      </View>
+      <View style={styles.details}>
+        <View style={styles.leftColumn}>
+          { showTime && <TimeAgo
+            date={timestamp}
+            component={DateText}
+            live={true}
+            formatter={timeFormatter}
+          /> }
+          <Sats
+            sats={transaction.type === TransactionType.Send ?
+              -transaction.sent :
+              transaction.received
+            }
+            currencyStyle={styles.currency}
+            satsStyle={[styles.sats, ! showTime && styles.satsNoTime]}
+            satsLabelStyle={styles.satsLabel}
+            usdStyle={styles.usd}
+            usdLabelStyle={styles.usdLabel}
+          />
+        </View>
+        <View style={styles.rightColumn}>
+          <AppText style={[styles.confirmations, confirmationsColorStyle]}>{ blockchainHeight && confirmationsText }</AppText>
+          <View style={styles.rightColumnBottom}>
+            <AppText numberOfLines={1} style={[styles.memo, ! memo && styles.noMemo]}>{ memo || 'No memo' }</AppText>
+            { transaction.address && <View style={styles.otherParties}>
+              <AppText style={styles.direction}>to</AppText>
+              <AppText style={styles.addressIO}>{formatAddress(transaction.address)}</AppText>
+            </View> }
+          </View>
+        </View>
+      </View>
+    </View>
+  );
+}
+
+function formatAddress(address: string): string {
+  if (address.length <= 16) {
+    return address;
+  }
+
+  // Show first and last eight characters - https://bitcoin.stackexchange.com/a/119182
+  const beginning = address.substring(0, 8);
+  const end = address.substring(address.length - 8, address.length);
+  return `${beginning}...${end}`;
+}
+
+function formatTime(date: Date): string {
+  return new Intl.DateTimeFormat('en-US', {
+    hour: 'numeric',
+    minute: 'numeric'
+  }).format(date).replace(' ', '').toLowerCase();
+}
+
+function formatDate(date: Date): string {
+  return new Intl.DateTimeFormat('en-US', {
+    month: 'short',
+    day: 'numeric',
+    year: 'numeric'
+  }).format(date);
+}
+
+function getConfirmations(currentBlockHeight: number, transactionBlockHeight?: number): number {
+  return transactionBlockHeight ?
+    currentBlockHeight - transactionBlockHeight + 1 :
+    0;
+}
+
+// returns one of: unconfirmed, 1 block deep, (2|3|4|5|6+|10+|100+|1k+|10k+|100k+) blocks deep
```

Comment:
If we want to add descriptions to functions, how about we use jsdoc?

Reply:
Done